dump-persistence-schema-compiler-plugin/src/main/scala/org/virtuslab/ash/DumpPersistenceSchemaCompilerPluginComponent.scala

```diff
@@ -5,21 +5,25 @@ import scala.tools.nsc.plugins.PluginComponent
 
 import org.virtuslab.ash.model.Field
 import org.virtuslab.ash.model.TypeDefinition
-import org.virtuslab.ash.writer.EventSchemaWriter
+import org.virtuslab.ash.writer.PersistenceSchemaWriter
 
-class DumpEventSchemaCompilerPluginComponent(val options: DumpEventSchemaOptions, val global: Global)
+class DumpPersistenceSchemaCompilerPluginComponent(val options: DumpPersistenceSchemaOptions, val global: Global)
     extends PluginComponent {
   import global._
-  override val phaseName: String = "dump-event-schema"
+  override val phaseName: String = "dump-persistence-schema"
   override val runsAfter: List[String] = List("typer")
 
-  lazy val writer = new EventSchemaWriter(options)
+  lazy val writer = new PersistenceSchemaWriter(options)
 
   override def newPhase(prev: Phase): Phase =
     new StdPhase(prev) {
 
       private val genericsNames =
```

Comment:
Okay, I'd clarify here that what all these types have in common is that:
- they have 2 or more type params,
- the last type param corresponds to state type,
- the last but one type param corresponds to event type.

Reply:
With that knowledge, understanding the logic in `foundUsedClasses` is much easier